docs: update troubleshooting for enterprise info (#48882)
gravitational/teleport: docs/pages/admin-guides/management/admin/troubleshooting.mdx

<TabItem scope={["oss"]} label="Teleport Community Edition">
If you need help, please ask on our [community forum](https://github.com/gravitational/teleport/discussions). You can also open an [issue on GitHub](https://github.com/gravitational/teleport/issues).

> For more information about Enterprise features reach out to us at [sales](https://goteleport.com/signup/enterprise/).

Reviewer: ```suggestion
For more information about Enterprise features reach out to [the Teleport sales team](https://goteleport.com/signup/enterprise/).
```
To make this a bit more specific

> You can also sign up for a [free trial](https://goteleport.com/signup) of Teleport Enterprise (Cloud).

Reviewer: ```suggestion
You can also sign up for a [free trial](https://goteleport.com/signup) of Teleport Enterprise.
```

I think in this case, it's okay to leave the deployment mode unspecified, since a user could still enjoy most self-hosted Enterprise features using the cloud-hosted deployment mode.
